packages/compiler/src/core/checker.ts

```diff
@@ -3605,6 +3605,12 @@ export function createChecker(program: Program): Checker {
       : declaration.parameters.length;
 
     if (args.length < minArgs || (maxArgs !== undefined && args.length > maxArgs)) {
+      // In the case we have too little args then this decorator is not applicable.
+      // If there is too many args then we can still run the decorator as long as the args are valid.
```

Comment:
Just thinking... folks doing this are kind of opting in for breaking changes. Should we consider not allowing this?

Reply:
so it is an error it will show up as an error and it will not be able to keep going past the checker. But this just means that we do a little bit of error recovery and let the decorators run and report any additional issues as there is nothing wrong from their side.

Reply:
I see I see.